platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/collections/AbstractLongList.java

```diff
@@ -627,8 +644,8 @@ private void checkValue(final long value) {
     /** {@inheritDoc} */
     @Override
     public <T extends Throwable> void forEach(final LongAction<T> action) throws InterruptedException, T {
-        final long max = size();
-        for (long i = minValidIndex.get(); i < max; i++) {
+        final long max = maxValidIndex.get();
+        for (long i = minValidIndex.get(); i <= max; i++) {
             final long value = get(i);
```

Comment:
BTW, this will spin the loop once in case `min == max == -1`. Not sure if it's intended.

Reply:
Indeed. If called for an empty list, this method would throw an exception. I added a check if the list is empty.